plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -170,6 +169,29 @@ public void testBucketNumberCompare()
         assertBucketAndHashEquals("timestamptz", LocalDateTime.of(1965, 1, 2, 13, 30, 15).toEpochSecond(UTC) * MICROSECONDS_PER_SECOND + 123_456, 236109233);  // before epoch
     }
 
+    /**
+     * Test example values from https://iceberg.apache.org/spec/#appendix-b-32-bit-hash-requirements
+     */
+    @Test
+    public void testBucketingSpecValues()
+    {
+        assertBucketAndHashEquals("int", 34, 2017239379);
+        assertBucketAndHashEquals("long", 34L, 2017239379);
+        assertBucketAndHashEquals("decimal(4, 2)", "14.20", -500754589 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("decimal(10, 2)", "14.20", -500754589 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("decimal(22, 2)", "14.20", -500754589 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("date", toIntExact(LocalDate.of(2017, 11, 16).toEpochDay()), -653330422 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("time", LocalTime.of(22, 31, 8).toNanoOfDay() / NANOSECONDS_PER_MICROSECOND, -662762989 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("timestamp", LocalDateTime.of(2017, 11, 16, 22, 31, 8).toEpochSecond(UTC) * MICROSECONDS_PER_SECOND, -2047944441 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("timestamptz", LocalDateTime.of(2017, 11, 16, 14, 31, 8).toEpochSecond(ZoneOffset.ofHours(-8)) * MICROSECONDS_PER_SECOND, -2047944441 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("string", "iceberg", 1210000089);
+        assertThatThrownBy(() -> assertBucketAndHashEquals("uuid", null /* f79c3e09-677c-4bbd-a479-3f349cb785e7 */, 1488055340))
+                // TODO https://github.com/trinodb/trino/issues/6663 Iceberg UUID should be mapped to Trino UUID and supported for bucketing
+                .hasMessage("Cannot convert from Iceberg type 'uuid' (UUID) to Trino type");
+        assertBucketAndHashEquals("fixed[4]", ByteBuffer.wrap(new byte[] {0x00, 0x01, 0x02, 0x03}), -188683207 & Integer.MAX_VALUE);
+        assertBucketAndHashEquals("binary", ByteBuffer.wrap(new byte[] {0x00, 0x01, 0x02, 0x03}), -188683207 & Integer.MAX_VALUE);
```

Comment:
This is not a per-spec value, but i believe this is actually a typo in the spec. https://github.com/apache/iceberg/pull/2840